### DISCUSSION TITLE
How to  limit cache size in HybridCache .

### DISCUSSION THREAD

**Discussion by madhub** [dotnet/aspnetcore#57582] [General]
Existing MemoryCache has option to set the cache limit via MemoryCacheOptions, if an application **already register** the MemoryCache with limit , all HybridCache  **API starts to fail** with exception "**'Cache entry must specify a value for Size when SizeLimit is set'**"

```csharp
// IMemoryCache limit
builder.Services.AddMemoryCache( options =>
{
    options.SizeLimit = 1024;
});


builder.Services.AddHybridCache(hybridCacheOptions =>{
    hybridCacheOptions.MaximumPayloadBytes = 1024 * 1024;
    hybridCacheOptions.MaximumKeyLength = 1024;
    hybridCacheOptions.DefaultEntryOptions = new HybridCacheEntryOptions
    {
        // set default cache expiration times to 1 minute.
        // it can be overriden by the per-call options if needed
        LocalCacheExpiration = TimeSpan.FromMinutes(1), // local in-memory cache expiration
        Expiration = TimeSpan.FromMinutes(2) // distributed cache expiration ( redis/valkey)
    };

```

**Comment by mgravell** [comment]
Great find! Will resolve ASAP. Note that we're relocating the HybridCache library to dotnet/extensions (for internal logistics etc) - the fix will be over there; I will cross-reference when possible.

**Comment by mgravell** [comment]
Tracking fix [over here](https://github.com/dotnet/extensions/pull/5420) - `HybridCache` has moved to dotnet/extensions
